spec/src/main/asciidoc/jakarta-ee.adoc

```diff
@@ -1,24 +1,42 @@
 == Jakarta Data Providers
 
-Jakarta Data providers implement repository interfaces, performing queries and other operations related to entities per the rules of the Jakarta Data specification. A Jakarta Data provider makes the repository implementation available to the application via dependency injection. The Jakarta Data specification defines the rules by which Jakarta Data providers must abide to ensure that multiple Jakarta Data providers are able to coexist in a system without interfering or overlapping on the same injection points.
+A Jakarta Data provider might come as an integrated component of a Jakarta EE container, or it might be a separate component that integrates with the Jakarta EE container via standard or proprietary SPIs.
+For example, a Jakarta Data provider might use a CDI portable extension to integrate with dependency injection.
+The role of the Jakarta Data provider is to supply implementations of application-defined repositories.
+
+A Jakarta Data provider must supply an implementation of each repository interface which either:
+
+- explicitly specifies the name of the provider via the `provider` member of the `@Repository` annotation, or
+- does not explicitly specify a `provider`, but whose entity classes are annotated with an entity-defining annotation type supported by the provider.
+
+The repository implementation, and the provider itself, is responsible for mediating interaction between the application program and the datastore, often taking advantage of other Jakarta EE services such as datasource provisioning and transaction management.
+
+An application usually obtains a reference to an implementation of a repository via dependency injection.
+Therefore, the provider must make its repository implementations available to the bean container responsible for dependency injection.
+
+Multiple Jakarta Data provider may coexist within a single program, and provide repository implementations.
+It is the responsibility of the application developer to ensure that each repository is eligible for implementation by exactly one provider, according to the rules specified below.
 
 === Provider support for Entities
 
-A Jakarta Data provider supplies an implementation of repository interfaces in Jakarta Data for one or more types of entities. An entity refers to a class that represents objects in a storage engine, such as SQL or NoSQL databases.
+A Jakarta Data provider supports one or more entity-defining annotation types.
```

Comment:
This is another subtle change that inadvertently rules out providers supporting unannotated entities.  We can adjust it as follows to preserve the meaning:
```suggestion
A Jakarta Data provider typically supports one entity-defining annotation type, but can support any number of entity-defining annotation types.
```

Reply:
You really want to allow providers which support _no_ entity-defining annotation?

OK, I guess.

Reply:
I remember it being discussed a while back, and wanted to be sure we weren't accidentally removing it.  I did do some experimenting with the idea myself and it did seem workable, albeit limiting.